### ISSUE TITLE
Changing IPv4 address with Powershell

### ISSUE THREAD

**Issue by dmidlo** [dmidlo/blog#5]
### Tldr

Get the `ifIndex`  or `Name` (alias) of the interface you want to change to IP address of:

```PowerShell
Get-NetAdapter | Format-Table
```

With the `New-NetIPAddress` cmdlet, Set the new IP address using the `ifIndex` or `Name` (alias) of your chosen interface.

```PowerShell
$InterfaceAlias = "Ethernet"
$IpAddress = "x.x.x.x"
$PrefixLength = 24
$DefaultGateway = "y.y.y.y"

New-NetIPAddress -InterfaceAlias $InterfaceAlias -IPAddress $IpAddress -PrefixLength $PrefixLength -DefaultGateway $DefaultGateway

# You can confirm the change with:
Get-NetIPAddress -InterfaceAlias $InterfaceAlias

# To remove an IP address from an interface
Remove-NetIPAddress -InterfaceAlias $InterfaceAlias
```

To set DNS servers:

```PowerShell
Set-DnsServerAddress -InterfaceAlias $InterfaceAlias -ServerAddresses "z.z.z.z, a.a.a.a"

# To confirm the change
Get-DnsServerAddress -InterfaceAlias $InterfaceAlias
```

#### Discussion

If you're curious or the need arises, you can list all your IP interfaces with the command `Get-NetIPAddress`.  This will give all the details of your interfaces, but it's poorly formatted and kinda hard to read.  So if you're just on the console and need to check use `Format-Table` with it.

```Powershell
Get-NetIPAddress | Format-Table
```

if you want to list just, say, IPv4 interfaces, you could do something like this:

```PowerShell
Get-NetIPAddress -AddressFamily IPv4 | Format-Table
```

You can also use `Get-NetIPConfiguration`:

```PowerShell
Get-NetIPConfiguration
```

You can also use `netsh`

```cmd
:: Show all interface configs
netsh interface ipv4 show config

:: Set the new address
set INTERFACE_ALIAS="Ethernet"
set IP_ADDRESS="x.x.x.x"
set MASK="m.m.m.m"
set GATEWAY="y.y.y.y"
netsh interface ipv4 set address name=%INTERFACE_ALIAS% static %IP_ADDRESS% %MASK% %GATEWAY%

:: Set DNS servers
set DNS1="z.z.z.z"
set DNS2="a.a.a.a"
netsh interface ipv4 set dns name=%INTERFACE_ALIAS% static %DNS1%
netsh interface ipv4 set dns name=%INTERFACE_ALIAS% static %DNS2% index=2

:: Show the config with netsh
netsh interface ipv4 show config %INTERFACE_ALIAS%

:: Set the interface to dhcp as well as the dns servers
netsh interface ipv4 set address name=%INTERFACE_ALIAS% source=dhcp
netsh interface ipv4 set dnsservers name=%INTERFACE_ALIAS% source=dhcp
```
